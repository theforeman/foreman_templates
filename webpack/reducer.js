import { combineReducers } from 'redux';

import syncSettings from './components/NewTemplateSync/NewTemplateSyncReducer';
import syncResult from './components/TemplateSyncResult/TemplateSyncResultReducer';

export default combineReducers({ syncSettings, syncResult });
