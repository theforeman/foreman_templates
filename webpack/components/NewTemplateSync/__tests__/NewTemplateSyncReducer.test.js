import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';

import reducer, { initialState } from '../NewTemplateSyncReducer';

import {
  importSettings,
  exportSettings,
} from '../__fixtures__/templateSyncSettings.fixtures';
import {
  SYNC_SETTINGS_REQUEST,
  SYNC_SETTINGS_SUCCESS,
  SYNC_SETTINGS_FAILURE,
} from '../../../consts';

const successPayload = {
  results: {
    import: importSettings,
    export: exportSettings,
  },
};

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },
  'should start loading on setting values request': {
    state: initialState,
    action: {
      type: SYNC_SETTINGS_REQUEST,
    },
  },
  'should stop loading on setting values success': {
    state: initialState.set('loadingSettings', true),
    action: {
      type: SYNC_SETTINGS_SUCCESS,
      payload: successPayload,
    },
    'should stop loading on setting values error': {
      state: initialState.set('loadingSettings', true),
      action: {
        type: SYNC_SETTINGS_FAILURE,
        payload: {
          error: 'Failed to fetch setting values',
        },
      },
    },
  },
};

describe('NewTemplateSyncReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
