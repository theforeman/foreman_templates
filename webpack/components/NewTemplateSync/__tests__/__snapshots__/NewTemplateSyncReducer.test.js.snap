// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTemplateSyncReducer should return initial state 1`] = `
Object {
  "error": "",
  "exportSettings": Array [],
  "importSettings": Array [],
  "loadingSettings": false,
}
`;

exports[`NewTemplateSyncReducer should start loading on setting values request 1`] = `
Object {
  "error": "",
  "exportSettings": Array [],
  "importSettings": Array [],
  "loadingSettings": true,
}
`;

exports[`NewTemplateSyncReducer should stop loading on setting values success 1`] = `
Object {
  "error": "",
  "exportSettings": Array [
    Object {
      "description": "Import or export names matching this regex (case-insensitive; snippets are not filtered)",
      "id": 47,
      "name": "filter",
      "settingsType": "string",
      "value": "",
    },
    Object {
      "description": "Negate the prefix/filter (for importing/exporting)",
      "id": 48,
      "name": "negate",
      "settingsType": "bool",
      "value": false,
    },
    Object {
      "description": "Target path to import and export. Different protocols can be used, for example /tmp/dir, git://example.com, https://example.com, ssh://example.com. When exporting to /tmp, note that production deployments may be configured to use private tmp.",
      "id": 49,
      "name": "repo",
      "settingsType": "string",
      "value": "https://github.com/theforeman/community-templates.git",
    },
  ],
  "importSettings": Array [
    Object {
      "description": "Associate templates to OS, organization and location",
      "id": 45,
      "name": "associate",
      "selection": Array [
        Object {
          "label": "New",
          "value": "new",
        },
        Object {
          "label": "Never",
          "value": "never",
        },
        Object {
          "label": "Always",
          "value": "always",
        },
      ],
      "settingsType": "string",
      "value": "new",
    },
    Object {
      "description": "Should importing overwrite locked templates?",
      "id": 46,
      "name": "force",
      "settingsType": "bool",
      "value": false,
    },
  ],
  "loadingSettings": false,
}
`;
